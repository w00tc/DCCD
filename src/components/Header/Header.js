import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';

import style from './Header.module.css';

import logo from '../../img/logo.png';

import { setStatus } from '../../redux/actions/items.actions';

export const Header = (props) => {
    const dispatch = useDispatch();
    const handleClick = () => {
        dispatch(setStatus(''));
    };
    return (
        <header className={style.headerSection}>
            <div className={style.containerFluid}>
                <div className={style.siteLogo}>
                    <img src={logo} alt=""/>
                </div>
                <ul className={style.menuList}>
                    <NavLink to="/">
                        <li>Home</li>
                    </NavLink>
                    <NavLink onClick={handleClick} to="/find">
                        <li>Cargo ID</li>
                    </NavLink>
                    <NavLink onClick={handleClick} to="/new">
                        <li>New Transaction</li>
                    </NavLink>
                    <NavLink to="/pendings">
                        <li>Pendings</li>
                    </NavLink>
                </ul>
            </div>
        </header>
    );
};

