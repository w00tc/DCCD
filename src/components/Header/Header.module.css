@font-face {
    font-family: 'Futura';
    src: url("../../fonts/Futura.woff");
}

.headerSection {
    padding: 15px 0;
    z-index: 99;
    color: #16d0c5;
}
.containerFluid {
    display: grid;
    position: relative;
    grid-auto-flow: column;
    margin: 0 10%;
}
.siteLogo {
    display: grid;
    align-content: center;
    justify-content: start;
}
.siteLogo a img {
    max-width: 100%;
}
.menuList {
    list-style: none;
    position: relative;
    display: grid;
    align-content: center;
    justify-content: end;
    grid-auto-flow: column;
    gap: 7%;
    padding-right: 6%;
}
.menuList li {
    display: inline;
    color: #002d76;
    font-weight: 500;
}
.menuList li a {
    display: inline-block;
    font-family: "Futura", sans-serif;
    font-size: 16px;
    color: #002d76;
    text-decoration: none;
    background-color: transparent;
}
.menuList li a.active {
    text-decoration: white;
    color: #002d76;
}
