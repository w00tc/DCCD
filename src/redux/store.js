import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import thunkMiddleware from 'redux-thunk';
import items from './reducers/items';
import pendings from './reducers/pendings';

const reducers = combineReducers({
    items,
    pendings,
});
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(reducers, composeEnhancers(applyMiddleware(thunkMiddleware)));
window.__store__ = store;
export default store;
