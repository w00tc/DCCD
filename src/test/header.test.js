import React from 'react';
import renderer from 'react-test-renderer';
import store from '../redux/store';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Header } from '../components/Header/Header';
import { BrowserRouter } from 'react-router-dom';

describe('With React Testing Library', () => {
    const mockStore = configureStore(store);

    it('Shows "Rendered Component"', () => {
        const component = renderer.create(<Provider store={store}><BrowserRouter>
            <Header/>
        </BrowserRouter></Provider>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        expect(component);
    });
});
