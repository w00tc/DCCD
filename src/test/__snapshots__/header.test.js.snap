// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`With React Testing Library Shows "Rendered Component" 1`] = `
<header
  className="headerSection"
>
  <div
    className="containerFluid"
  >
    <div
      className="siteLogo"
    >
      <img
        alt=""
        src="logo.png"
      />
    </div>
    <ul
      className="menuList"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <li>
          Home
        </li>
      </a>
      <a
        aria-current={null}
        href="/find"
        onClick={[Function]}
      >
        <li>
          Cargo ID
        </li>
      </a>
      <a
        aria-current={null}
        href="/new"
        onClick={[Function]}
      >
        <li>
          New Transaction
        </li>
      </a>
      <a
        aria-current={null}
        href="/pendings"
        onClick={[Function]}
      >
        <li>
          Pendings
        </li>
      </a>
    </ul>
  </div>
</header>
`;
